consensus/wal_generator.go

```diff
@@ -89,7 +89,7 @@ func WALGenerateNBlocks(t *testing.T, wr io.Writer, numBlocks int) (err error) {
 	consensusState := NewState(config.Consensus, state.Copy(), blockExec, blockStore, mempool, evpool)
 	consensusState.SetLogger(logger)
 	consensusState.SetEventBus(eventBus)
-	if privValidator != nil {
+	if privValidator != nil && privValidator != (*privval.FilePV)(nil) {
```

Comment:
???

Reply:
I was trying just checking nil for "no validator", but there were cases `privValidator != nil` even though privValidator is empty with `ModeFullNode`, maybe interface typecasting issue 

so originally I added the Empty() function to the interface `PrivValidator` for checking "no validator"
```
func (pv *FilePV) Empty() bool {
    return pv == (*FilePV)(nil)
}
```    

but then changed it simply to using config because there were too many relevant correction codes.

https://github.com/tendermint/tendermint/pull/5212#discussion_r487417366